File: src/controllers/auth/signin.controllers.js
import * as signinService from "../../services/auth/signin.service.js"

export const signin = async (req, res) => {
  try {
    const { email_or_username, password: inputPassword } = req.body

    const resp = await signinService.signin(email_or_username, inputPassword)

    if (resp.error) return res.status(404).send(resp.error)

    req.session.user = { authJwt: resp.jwt }
    
    req.sessionOptions.path = "/"
    req.sessionOptions.maxAge = 10 * 24 * 60 * 60 * 1000

    res.status(200).send(resp.data)
  } catch (error) {
    console.error(error)
    res.sendStatus(500)
  }
}

File: src/controllers/auth/signinControllers/signinControllers.go
package signinControllers

import (
	"i9lyfe/src/helpers"
	"i9lyfe/src/services/auth/signinService"
	"time"

	"github.com/gofiber/fiber/v3"
)

func Signin(c fiber.Ctx) error {
	ctx := c.Context()

	var body signInBody

	err := c.Bind().MsgPack(&body)
	if err != nil {
		return err
	}

	if err := body.Validate(); err != nil {
		return err
	}

	respData, authJwt, err := signinService.Signin(ctx, body.EmailOrUsername, body.Password)
	if err != nil {
		return err
	}

	reqSession := map[string]any{
		"user": map[string]any{"authJwt": authJwt},
	}

	c.Cookie(helpers.Session(reqSession, "/api/app", int(10*24*time.Hour/time.Second)))

	return c.MsgPack(respData)
}
